src/commands/cmd_bulk_insert.c

```diff
@@ -10,24 +10,53 @@
 #include "../util/thpool/pools.h"
 #include "../bulk_insert/bulk_insert.h"
 
-// bulk-insert context object
-typedef struct {
-	int argc;                      // number of arguments
-	RedisModuleString **argv;      // arguments
-	RedisModuleBlockedClient *bc;  // blocked client
-} BulkCtx;
+#define BATCH_SIZE_LIMIT 1000000 // 1M
+
+// process HEADER token if exists
+static int _Graph_Bulk_Header(RedisModuleCtx *ctx, RedisModuleString ***argv,
+		int *argc, Graph *g) {
+	const char *token = RedisModule_StringPtrLen(**argv, NULL);
+	bool header = strcmp(token, "HEADER") == 0;
+	if(header) {
+		// skip header token
+		(*argv) ++;
+		(*argc) --;
+
+		// HEADER:
+		// total node count
+		// total edge count
+
+		long long node_count = 0;  // number of nodes in graph
+		long long edge_count = 0;  // number of edges in graph
+
+		if(RedisModule_StringToLongLong(**argv++, &node_count) != REDISMODULE_OK) {
```

Comment:
This will not behave as desired, we have to invoke `**argv` here and `(*argv)++` afterward (or use bracket subscripts).

Reply:
👍